### PR TITLE
fix(primary): fix primary name request creation, replacing name for existing owner

### PR DESCRIPTION
**Pull request by dtfiedler**
Allows a user to replace an existing primary name request with a new one, so long as it is a different name

### CODE REVIEW

File: src/primary_names.lua [hunk 1]
```diff
@@ -58,9 +58,13 @@ function primaryNames.createPrimaryNameRequest(name, initiator, timestamp, msgId
...
 	local baseName = baseNameForName(name)
 
-	--- existing request for primary name from wallet?
-	local existingRequest = primaryNames.getPrimaryNameRequest(name)
-	assert(not existingRequest, "Primary name request for '" .. name .. "' already exists") -- TODO: should we error here or just let them create a new request and pay the fee again?
+	--- check the primary name request for the initiator does not already exist for the same name
+	--- this allows the caller to create a new request and pay the fee again, so long as it is for a different name
+	local existingRequest = primaryNames.getPrimaryNameRequest(initiator)
+	assert(
+		not existingRequest or existingRequest.name ~= name,
```
Comment (arielmelendez) [inline src/primary_names.lua:65]: Is any normalization needed for the names? Casing etc?

File: src/primary_names.lua [hunk 2]
```diff
@@ -58,9 +58,13 @@ function primaryNames.createPrimaryNameRequest(name, initiator, timestamp, msgId
...
+		"Primary name request for '" .. initiator .. "' for '" .. name .. "' already exists"
```
Comment (arielmelendez) [inline src/primary_names.lua:66]: Nit: "Primary name request `BY` '"

**Review by arielmelendez — APPROVED** [review]

**Review by arielmelendez — DISMISSED** [review]

File: src/primary_names.lua (hunk 1 above)
Comment (dtfiedler) [reply src/primary_names.lua:65]: we lower case all the names on the original request - can add same logic here jic

**Review by fedellen — DISMISSED** [review]

File: src/primary_names.lua (hunk 2 above)
Comment (dtfiedler) [reply src/primary_names.lua]: [2c49ac2](https://github.com/ar-io/ar-io-network-process/pull/230/commits/2c49ac2d168e7024a09f6c48d821447ccad212af)

File: src/primary_names.lua (hunk 1 above)
Comment (dtfiedler) [reply src/primary_names.lua:66]: enforced it again here - [2c49ac2](https://github.com/ar-io/ar-io-network-process/pull/230/commits/2c49ac2d168e7024a09f6c48d821447ccad212af)

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**